projects/reservoir/fv3/era_process.py

```diff
@@ -212,6 +243,62 @@ def mask(coarsened):
     return masked
 
 
+def calculate_weekly_mean(sorted_data):
+    weekly_merged = xr.concat(
+        [
+            sorted_data.isel(time=slice(i * 7, (i + 1) * 7)).mean("time")
+            for i in range(int(sorted_data.time.shape[0] / 7))
+        ],
+        dim="time",
+    )
+    n = len(weekly_merged["time"])
+    weekly_merged["time"] = sorted_data["time"][::7][:n]
+    return weekly_merged
+
+
+def shift_variable(data):
+    # rename variable
+    return data.shift(time=-1)
+
+
+def calculate_sst_tendency(args):
+    seven_days_in_seconds = 604800
+    sst_dataset = xr.open_dataset(
+        args.out_path + "sst_" + args.train_val_test + "_regridded.nc",
+        decode_times=False,
+    )
+    sst_resid_data_set = sst_dataset.copy()
+    sst_resid_data_set = (
+        sst_resid_data_set.shift(time=-1) - sst_resid_data_set
+    ) / seven_days_in_seconds
+    # rename dataset
+    sst_resid_data_set = sst_resid_data_set.rename_vars({"sst": "sst_tendency"})
+    # save residual dataset
+    sst_resid_data_set.to_netcdf(
+        args.out_path + "sst_tendency_" + args.train_val_test + "_regridded.nc"
+    )
+
+
+def merge_variables(variables, args):
+    data_list = []
+    for var in variables:
+        # load data
+        data = xr.open_dataset(
+            args.out_path + var + "_" + args.train_val_test + "_regridded.nc",
+            decode_times=False,
+        )
+        data_list.append(data)
+    return xr.merge(data_list)
+
+
+def save_data_as_tiles(data, args):
+    # save each tile individually
+    for i in range(len(data.tile)):
+        data.isel(tile=i).to_netcdf(
```

Comment:
The data loaders that are used in training take in a directory which is assumed to contain all the netcdfs for a single rank (tile). Could you change the output path to match this structure (one tile data per directory)?